java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -602,6 +606,169 @@ public static CompletableFuture<Object> arrinsert(
                         .toArray());
     }
 
+    /**
+     * Searches for the first occurrence of a <code>scalar</code> JSON value in the arrays at the
+     * path.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @param path The path within the JSON document.
+     * @param scalar The scalar value to search for; JSON scalar refers to values that are not objects
+     *     or arrays.
+     * @return
+     *     <ul>
+     *       <li>For JSONPath (<code>path</code> starts with <code>$</code>): Returns an array with a
+     *           list of integers for every possible path, indicating the index of the matching
+     *           element. The value is <code>-1</code> if not found. If a value is not an array, its
+     *           corresponding return value is <code>null</code>.
+     *       <li>For legacy path (path doesn't start with <code>$</code>): Returns an integer
+     *           representing the index of matching element, or <code>-1</code> if not found. If the
+     *           value at the <code>path</code> is not an array, an error is raised.
+     *     </ul>
+     *
+     * @example
+     *     <pre>{@code
+     * Json.set(client, key, "$", "{\"a\": [\"value\", 3], \"b\": {\"a\": [3, [\"value\", false], 5]}}").get();
+     * var result = Json.arrindex(client, key, "$..a", new JsonScalar(3)).get();
+     * assert Arrays.equal((Object[]) result, new Object[] {1L, 0L});
```

Comment:
I'd suggest that we copy-paste this and make sure it compiles. We could also add it to a test and make sure it's correct.  
```suggestion
     * assert Arrays.equals((Object[]) result, new Object[] {1L, 0L});
```